### DISCUSSION TITLE
Heltec esp32 lora v2

### DISCUSSION THREAD

**Discussion by Guldtistel** [jgromes/RadioLib#872] [Q&A]
Hi 

Has anyone got lora to work on a "Heltec esp32 lora V2" with Radiolib? If someone got it to work, what settings do you use? I get everything to work with a "ttgo esp32 lora", but not with Heltec.

**Comment by RonMeier** [comment]
maybe you can try the following:

Heltec32 V2:    
SX1276 radio = new Module(18,26,14,5); 
Heltec32 V3:     
SX1262 radio = new Module(8,14,12,13);

**Comment by Guldtistel** [comment]
Thank you for your reply.
 Now I've tried more. Between Heltec to Heltec it works well, between Ttgo to Ttgo it is also fine, but between Heltec and Ttgo it does not work! It's the same pinout on both cards, and the same program, a transmitter program and a receiver program. When you send between the same make, it never misses, but if you send between different brands, it misses 95% of the receptions. What would I change to make it work?

**Comment by RonMeier** [comment]
However, operation under different manufacturers should work. I have Helec and ttgo mixed together and they all work very well with each other.  As a suggestion, I would explicitly define the same parameters in the transmitter and receiver in Setup(), maybe that will help.

    radio.setFrequency(868.0); 
    radio.setSpreadingFactor(7);     
    radio.setSyncWord(18);      
    radio.setBandwidth(125.0);
    radio.setCodingRate(5); 
    radio.setPreambleLength(8);
    radio.invertIQ(0);                            
    radio.setCRC(0);

**Comment by Guldtistel** [comment]
This is beyond my expertise, but when I changed
spread factor from 7 to 6 on both it started working. Why?
The value of Heltec to Heltec is RSSI-90dbm and SF 10db
The value of Heltec to TTgo is RSSI-88dbm and SF 2db
They are about 2 meters apart.
Is it acceptable if you need approx. 200m range?

Thanks so much for your help.

**Comment by RonMeier** [comment]
What is important is always the same SF at the sender and receiver. I would start with 7, and if the connection isn't too good,
slowly go higher. However, the higher the SF, the more power the devices require. Specified is 6-12.
A distance of 200m with a reasonably good antenna should be easily manageable, I can cover 2-3km in a built-up area without any problems.
You also have the option of specifying the transmitter power: 
maximum power:  radio.setOutputPower(23);
